### DISCUSSION TITLE
Service without hash

### DISCUSSION THREAD

**Discussion by larivierec** [envoyproxy/gateway#2922] [Q&A] [answered]
Hello,

I just recently installed the gateway-helm and so far it's working great.
One comment I have and this is really a nitpick, is there a way to remove the generated hash at the end of the kubernetes service?

```bash
(⎈|k3s:ingress)➜  home-cluster git:(main) ✗ k -n ingress get svc

NAME                                                  TYPE           CLUSTER-IP      EXTERNAL-IP     PORT(S)                      AGE
envoy-gateway                                         ClusterIP      10.43.210.20    <none>          18000/TCP,18001/TCP          31m
envoy-gateway-metrics-service                         ClusterIP      10.43.142.165   <none>          19001/TCP                    31m
envoy-infrastructure-envoy-external-119c46e1          LoadBalancer   10.43.77.221    192.168.40.22   80:31231/TCP,443:31369/TCP   31m
envoy-infrastructure-envoy-internal-bfbc22f7          LoadBalancer   10.43.173.168   192.168.40.23   80:31589/TCP,443:32523/TCP   31m
ingress-external-ingress-nginx-controller             LoadBalancer   10.43.101.29    192.168.40.3    80:30253/TCP,443:30680/TCP   27d
ingress-external-ingress-nginx-controller-admission   ClusterIP      10.43.145.33    <none>          443/TCP                      27d
ingress-external-ingress-nginx-controller-metrics     ClusterIP      10.43.36.234    <none>          10254/TCP                    27d
ingress-internal-ingress-nginx-controller             LoadBalancer   10.43.159.75    192.168.40.2    80:30192/TCP,443:31564/TCP   189d
ingress-internal-ingress-nginx-controller-admission   ClusterIP      10.43.156.66    <none>          443/TCP                      189d
ingress-internal-ingress-nginx-controller-metrics     ClusterIP      10.43.108.102   <none>          10254/TCP                    189d
```

for example, if i want to use this with cloudflared, i'll have to put in the configuration the hash and if ever it regenerates the service it will break external services access.

example cloudflared configuration when not exposing ports to the outside world:

```yaml
    - hostname: "*.garb.dev"
      service: https://envoy-infrastructure-envoy-external-119c46e1.ingress.svc.cluster.local:443
      originRequest:
        originServerName: ingress.garb.dev
``` 

if this won't change by itself, we can resolve the discussion, but if it changes based on time, flux reconcile or hash of source code it may be an issue

**Comment by arkodg** [comment, marked as answer]
you should now be able to set a custom service name, more in https://github.com/envoyproxy/gateway/issues/2141#issuecomment-1972484035

**Reply by larivierec** [reply]
How would this work if I'm using two controllers
I.e internal and external?

I need two gateway classes?

https://github.com/larivierec/home-cluster/blob/main/kubernetes/apps/ingress/gateway/envoy/manifests/gateway-infrastructure.yaml

Here's my manifest if you're interested

**Reply by arkodg** [reply]
this is a field within the `EnvoyProxy` resource
https://gateway.envoyproxy.io/v1.0.0/api/extension_types/#kubernetesservicespec
